src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -528,95 +549,350 @@ internal override void Complete()
+
+                    // Capture screen output if we are transcribing
+                    if (this.Command.Context.EngineHostInterface.UI.IsTranscribing &&
+                        _scrapeHostOutput)
+                    {
+                        Host.Coordinates endPosition = this.Command.Context.EngineHostInterface.UI.RawUI.CursorPosition;
+                        endPosition.X = this.Command.Context.EngineHostInterface.UI.RawUI.BufferSize.Width - 1;
+
+                        // If the end position is before the start position, then capture the entire buffer.
+                        if (endPosition.Y < _startPosition.Y)
+                        {
+                            _startPosition.Y = 0;
+                        }
+
+                        Host.BufferCell[,] bufferContents = this.Command.Context.EngineHostInterface.UI.RawUI.GetBufferContents(
+                            new Host.Rectangle(_startPosition, endPosition));
+
+                        StringBuilder lineContents = new StringBuilder();
+                        StringBuilder bufferText = new StringBuilder();
+
+                        for (int row = 0; row < bufferContents.GetLength(0); row++)
+                        {
+                            if (row > 0)
+                            {
+                                bufferText.Append(Environment.NewLine);
+                            }
 
-                                bufferText.Append(lineContents.ToString().TrimEnd(Utils.Separators.SpaceOrTab));
+                            lineContents.Clear();
+                            for (int column = 0; column < bufferContents.GetLength(1); column++)
+                            {
+                                lineContents.Append(bufferContents[row, column].Character);
                             }
 
-                            this.Command.Context.InternalHost.UI.TranscribeResult(bufferText.ToString());
+                            bufferText.Append(lineContents.ToString().TrimEnd(Utils.Separators.SpaceOrTab));
                         }
 
-                        this.Command.Context.SetVariable(SpecialVariables.LastExitCodeVarPath, _nativeProcess.ExitCode);
-                        if (_nativeProcess.ExitCode != 0)
-                            this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
+                        this.Command.Context.InternalHost.UI.TranscribeResult(bufferText.ToString());
                     }
+
+                    this.Command.Context.SetVariable(SpecialVariables.LastExitCodeVarPath, _nativeProcess.ExitCode);
+                    if (_nativeProcess.ExitCode != 0)
+                        this.commandRuntime.PipelineProcessor.ExecutionFailed = true;
                 }
             }
             catch (Win32Exception e)
             {
-                exceptionToRethrow = e;
+                _exceptionToRethrow = e;
             } // try
             catch (PipelineStoppedException)
```

Comment:
Maybe. I though that `CommandProcessorBase.CheckForSevereException` is valuable check, but it may be not.